spec/unit/rom/gateway_spec.rb

```diff
@@ -82,9 +85,33 @@ class Gateway
 
   describe '#disconnect' do
     it 'does nothing' do
-      gateway_class = Class.new(ROM::Gateway)
-      gateway = gateway_class.new
       expect(gateway.disconnect).to be(nil)
     end
   end
+
+  describe '#logger' do
+    it 'does nothing' do
+      expect(gateway.logger).to be(nil)
+    end
+  end
+
+  describe '#use_logger' do
+    it 'does nothing' do
+      expect(gateway.use_logger).to be(nil)
+    end
+
+    it 'accepts parameters' do
+      params = [5, "mutations", "are", "nice"]
+
+      expect { gateway.use_logger(params) }.to_not raise_error
+    end
+  end
+
+  describe '#extend_command_class' do
+    it 'returns the class sent as 1st parameter' do
+      klass = ROM::Relation
+
+      expect(gateway.extend_command_class(klass, "foo")).to eq(klass)
```

Comment:
it would be nice to create an anonymous class and freeze it to make sure it simply returns the original one without changing it :)